### PR TITLE
blockheight bug

### PR DESCRIPTION
**Pull request by niftynei**
At cold start, if your node is behind the blocktip and you've sent your peer a blockheight counter from the future, we shouldn't confuse ourselves with our rollback/replay.

Should fix flakes in CI that were spotting BROKEN blockheight updates.

Also includes small patch to use the same GRPC paths for the `rm` cleanup.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -369,21 +369,22 @@ ifneq ($(RUST),0)
...
-	python -m grpc_tools.protoc -I cln-grpc/proto cln-grpc/proto/primitives.proto --python_out=contrib/pyln-grpc-proto/pyln/grpc/ --experimental_allow_proto3_optional
+	python -m grpc_tools.protoc -I cln-grpc/proto cln-grpc/proto/node.proto --python_out=$(GRPC_PATH)/ --grpc_python_out=$(GRPC_PATH)/ --experimental_allow_proto3_optional
+	python -m grpc_tools.protoc -I cln-grpc/proto cln-grpc/proto/primitives.proto --python_out=$(GRPC_PATH)/ --experimental_allow_proto3_optional
 	# The compiler assumes that the proto files are in the same
 	# directory structure as the generated files will be. Since we
 	# don't do that we need to path the files up.
-	find contrib/pyln-grpc-proto/pyln/ -type f -name "*.py" -print0 | xargs -0 sed -i'.bak' -e 's/^import \(.*\)_pb2 as .*__pb2/from pyln.grpc import \1_pb2 as \1__pb2/g'
-	rm -f contrib/pyln-grpc-proto/pyln/*.py.bak
+	find $(GRPC_PATH)/ -type f -name "*.py" -print0 | xargs -0 sed -i'.bak' -e 's/^import \(.*\)_pb2 as .*__pb2/from pyln.grpc import \1_pb2 as \1__pb2/g'
+	rm -f $(GRPC_PATH)/*.py.bak
```
Comment (rustyrussell) [inline Makefile:387]: So this last line is the real fix, since the find kinda worked, but the rm didn't?

**Review by rustyrussell — APPROVED** [review]
Ack bb2c900d0e96a706383eb299c70456df0e32d469

File: Makefile (hunk 1 above)
Comment (niftynei) [reply Makefile:387]: Oh no, I missed that the `find` uses a different path (up one level)... and there's even a comment, oops.

seems like the real fix would be to have `rm` do a recursive hunt also? ok let me try again here. rip.
